### PR TITLE
CLN/DEPR: remove deprecated pandas.rpy module (GH9602)

### LINKED ISSUE #9602: DEPR/DOC: rpy deprecation
so we should put up a deprecation notice on `rpy`imports that in the future the user should use  `rpy2` directly rather than the less-supported  `pandas.rpy` converters that we can take action say in 0.17.0

discussed extensivley in #7385 

xref #7385, #9187 #7309, #8255 
- [x] Update the docs with some example of how to port your pandas examples to the `rpy2` API -> done with #10075
- [ ] Go over the functionality in pandas and check if some things should be ported to `rpy2` (what still works better in pandas?)
  - [ ] Handling of NaN values (comment below: https://github.com/pydata/pandas/issues/9602#issuecomment-94716332)
  - [ ] index is not preserved in `ri2py` -> this should be handled in https://bitbucket.org/rpy2/rpy2/issues/285
- [ ] The closed PRs -> push them to `rpy2`?

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #9602

### CODE REVIEW

File: doc/source/r_interface.rst [hunk 1]
```diff
@@ -60,7 +60,7 @@ can be replaced with::
 
     from rpy2.robjects import r
     r.data('iris')
```
Comment (jreback) [inline doc/source/r_interface.rst]: I would remove everything here except for the top warning, then put a link to the previous page of the built docs.

**Comment by jreback** [comment]
just a doc comment, otherwise lgtm.

File: doc/source/r_interface.rst (hunk 1 above)
Comment (jorisvandenbossche) [inline doc/source/r_interface.rst]: Yep, the docs were still 'to do'. What I was thinking to do is update the current examples to just use the corresponding `rpy2` functions. So then there is still some documentation on how to use it (for the updating, I can refer to docs of previous version).

Comment (jreback) [inline doc/source/r_interface.rst]: sure if that works, just make a note (e.g. import at the top and indicate), you are importing an external library to do this.

**Comment by jreback** [comment]
good to go?

**Comment by sinhrks** [comment]
it's nice to update ecosystem doc to describe rpy2 and PypeR.

**Comment by jorisvandenbossche** [comment]
@sinhrks do you know how maintained PypeR is, and whether pandas is supported well or not?

**Comment by sinhrks** [comment]
@jorisvandenbossche their website is below, but can't find actual repo (looks to be not maintainanced 2014)? 

http://www.webarray.org/softwares/PypeR/

The package has ``use_pandas`` to handle ``DataFrame`` and ``Series``. From their source impl, ``rpy2`` is expected to work better.

```
        use_pandas: Used as a boolean value. A False value will disable pandas
            even if it has been imported. 
```

Pls drop ``PypeR`` description if it is not appropriate.

**Comment by jreback** [comment]
http://pandas-docs.github.io/pandas-docs-travis/r_interface.html#rpy

something not defined here?

**Comment by jorisvandenbossche** [comment]
Hmm, sigh. This worked locally, but I now see this only worked because I didn't activate pandas2ri. But if I don't activate pandas2ri, the second example does not work. I can't really get my head around rpy2's conversion/activate/dispatch rules and object model .. 
(will do a PR to fix this)

**Pull request merged**